src/common/operator/cast_operators.cpp

```diff
@@ -1482,6 +1492,30 @@ bool TryCastToBit::Operation(string_t input, string_t &result, Vector &result_ve
 	return true;
 }
 
+template <>
+bool CastFromBitToNumeric::Operation(string_t input, bool &result, bool strict) {
+	D_ASSERT(input.GetSize() > 1);
+
+	uint8_t value;
+	CastFromBitToNumeric::Operation(input, value, strict);
```

Comment:
Isn't returning the boolean instead of asserting it a better idea? Down the line that boolean gets used by the TryCastLoop.

Like this:
```cpp
uint8_t value;
bool success = CastFromBitToNumeric::Operation(input, value, strict);
result = (value > 0);
return (success);
```

I don't believe this return value should be asserted.